kernel/indTyping.ml

```diff
@@ -302,7 +303,19 @@ let abstract_packets ~template_check univs usubst params ((arity,lc),(indices,sp
           | Polymorphic _ ->
             CErrors.anomaly ~label:"polymorphic_template_ind"
               Pp.(strbrk "Template polymorphism and full polymorphism are incompatible.") in
-      let param_levels, concl_levels = template_polymorphic_univs ~template_check ctx params min_univ in
+      let ctor_levels = Array.fold_left
+          (fun levels (d,c) ->
+             List.fold_left (fun levels d ->
+                 Context.Rel.Declaration.fold_constr (fun c levels ->
+                     Vars.with_universes_of_constr levels c)
```

Comment:
Why not simply a union instead of exporting a new function? This is very unlikely to be performance-critical.

Reply:
because I got tired of typing these damn unions in the middle of writing the patch

Reply:
Come on, write a union and be done with it.